### ISSUE TITLE
Updating to 10.0.2 - OC\RepairException: Upgrade is not possible

### ISSUE THREAD

**Issue by HandeTech** [owncloud/core#29055]
### Steps to reproduce
1.
2.
3.

### Expected behaviour
Tell us what should happen

### Actual behaviour
Tell us what happens instead

### Server configuration
**Operating system**:

**Web server:**

**Database:**

**PHP version:**

**ownCloud version:** (see ownCloud admin page)

**Updated from an older ownCloud or fresh install:**

**Where did you install ownCloud from:**

**Signing status (ownCloud 9.0 and above):**

```
Login as admin user into your ownCloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results into https://gist.github.com/ and puth the link here.
```

**The content of config/config.php:**

```
Log in to the web-UI with an administrator account and click on
'admin' -> 'Generate Config Report' -> 'Download ownCloud config report'
This report includes the config.php settings, the list of activated apps
and other details in a well sanitized form.

or 

If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your ownCloud installation folder

*ATTENTION:* Do not post your config.php file in public as is. Please use one of the above
methods whenever possible. Both, the generated reports from the web-ui and from occ config:list
consistently remove sensitive data. You still may want to review the report before sending.
If done manually then it is critical for your own privacy to dilligently
remove *all* host names, passwords, usernames, salts and other credentials before posting.
You should assume that attackers find such information and will use them against your systems.
```

**List of activated apps:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your ownCloud installation folder.
```

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

#### LDAP configuration (delete this part if not used)

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your ownCloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```

### Client configuration
**Browser:**

**Operating system:**

#### Web server error log
```
Insert your webserver log here
```

#### ownCloud log (data/owncloud.log)
```
Insert your ownCloud log here
```

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log 
c) ...
```
***************************************************************
The following is the result after using cPanel and Softaculous updating attempt:

Updating to 10.0.2

OC\RepairException: Upgrade is not possible

        Repair warning: Exception
        Repair warning: Archives of type application/octet-stream are not supported
        Repair warning: Exception
        Repair warning: Archives of type application/octet-stream are not supported
        Repair warning: Exception
        Repair warning: Archives of type application/octet-stream are not supported
        Repair warning: Exception
        Repair warning: Archives of type application/octet-stream are not supported
        Repair warning: You have incompatible or missing apps enabled that could not be found or updated via the marketplace.
        Repair warning: please install app manually with tarball or disable them with: occ app:disable activity occ app:disable files_pdfviewer occ app:disable files_texteditor occ app:disable gallery

Detailed logs
Preparing update
Set log level to debug
Turned on maintenance mode
Repair warning: Exception
Repair warning: Archives of type application/octet-stream are not supported
Repair warning: Exception
Repair warning: Archives of type application/octet-stream are not supported
Repair warning: Exception
Repair warning: Archives of type application/octet-stream are not supported
Repair warning: Exception
Repair warning: Archives of type application/octet-stream are not supported
Repair warning: You have incompatible or missing apps enabled that could not be found or updated via the marketplace.
Repair warning: please install app manually with tarball or disable them with: occ app:disable activity occ app:disable files_pdfviewer occ app:disable files_texteditor occ app:disable gallery
OC\RepairException: Upgrade is not possible
The update was unsuccessful. Please report this issue to the ownCloud community.

**Comment by kdslkdsaldsal** [comment]
Looks similar to https://github.com/owncloud/core/issues/28079

You could also try to update to the latest 10.0.3 (releases a few days ago) instead of the older 10.0.2: https://owncloud.org/changelog/#latest10.0

**Comment by patrickjahns** [comment]
@HandeTech 

Can you run the upgrade command again with `-vvv` enabled and provide the console output?

**Comment by bogorya** [comment]
I recently tried to upgrade, first to 10.0.2 and then 10.0.3 - in both cases the error is the same:

Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'ALTER TABLE ujph_calendarsubscriptions CHANGE `lastmodified` lastmodified INT UNSIGNED DEFAULT NULL NOT NULL': SQLSTATE[42000]: Syntax error or access violation: 1067 Invalid default value for 'lastmodified'

It's obvious that SQL is faulty. It should be:
ALTER TABLE ujph_calendarsubscriptions CHANGE `lastmodified` lastmodified INT UNSIGNED DEFAULT NULL
or
ALTER TABLE ujph_calendarsubscriptions CHANGE `lastmodified` lastmodified INT UNSIGNED DEFAULT NOT NULL

I don't know what should be the default value for this field.

Piotr

**Comment by kdslkdsaldsal** [comment]
@bogorya This looks completely unrelated as there is none of such SQL errors within the initial report.

@HandeTech This might help: https://github.com/owncloud/core/issues/28079#issuecomment-333485833

> I eventually solved this in cPanel 66.0.23 / Installatron. I looked at the PHP settings in cPanel and ticked the box for the extension 'intl'.

Seems the ``intl`` module is only listed as optional here:

https://doc.owncloud.org/server/latest/admin_manual/installation/source_installation.html#optional

but might be possible that this is now required.

**Comment by HandeTech** [comment]
Update: 
This ownCloud install is the free/Community version. I have tried the upgrade path internally via ownCloud admin and via Softaculous in cPanel.

The issue I faced was on a Shared Host and the support staff there weren't much help as ownCloud is a "...3rd party app."

I ended up restoring the backup successfully yet again and that's where the situation will stay as I don't have the time, nor inclination to sort this upgrade issue out. This ownCloud install is for private/personal use only so the issue is trivial.

Current thinking is that I'll test NextCloud and migrate ownCloud going forward.

On a side note. With regard to cPanel installs, I seem to have more issues using Softaculous (failed or borked installs) whereas Fantastico runs like a charm.

@kdslkdsaldsal - thought this would be a winner but after restoring from backup, upgrade to 10.0.3 failed also. Stuck in maintenance mode.

@patrickjahns - not using a CLI, sorry.

@bogorya - Issue with MariaDB eh?
I wonder if that's why my hosting support staff installed a new instance using SQLite instead of the normal mySQL/MariaDB defaults. This of course was no help as all my files/pics were gone. Gotta luv backups!

Thanks for your input everyone.
